src/rgw/rgw_auth_s3.cc

```diff
@@ -346,16 +346,20 @@ int rgw_calculate_s3_v4_aws_signature(struct req_state *s,
     const string& access_key_id, const string &date, const string& region,
     const string& service, const string& string_to_sign, string& signature) {
 
-  map<string, RGWAccessKey>::iterator iter = s->user->access_keys.find(access_key_id);
-  if (iter == s->user->access_keys.end()) {
-    ldout(s->cct, 10) << "ERROR: access key not encoded in user info" << dendl;
-    return -EPERM;
-  }
-
-  RGWAccessKey& k = iter->second;
+  string secret_key;
+  if (s->aws4_auth->secret_access_key.empty()) {
+    map<string, RGWAccessKey>::iterator iter = s->user->access_keys.find(access_key_id);
+    if (iter == s->user->access_keys.end()) {
+      ldout(s->cct, 10) << "ERROR: access key not encoded in user info" << dendl;
+      return -EPERM;
```

Comment:
the statement is in a block, and should be 2spcs to the right of the if, as shown?